File: argon-plugins-tube/src/main/scala/dev/argon/plugins/tube/TubeOutput.scala
package dev.argon.plugins.tube

import dev.argon.io.ZipStreamResource
import dev.argon.options.OutputHandler

final case class TubeOutput[-R, +E]
(
  implementationModule: ZipStreamResource[R, E],
  interfaceModule: ZipStreamResource[R, E],
)

object TubeOutput:
  given [R, E]: OutputHandler[R, E, TubeOutput[R, E]] =
    OutputHandler.derive
end TubeOutput

File: plugin/tube/java/src/main/java/dev/argon/plugins/tube/TubeOutput.java
package dev.argon.plugins.tube;

import dev.argon.plugin.api.BinaryResource;
import dev.argon.plugin.api.options.Options;
import dev.argon.plugin.api.options.OptionsDecoding;
import org.jetbrains.annotations.NotNull;

@Options(decoding = OptionsDecoding.OUTPUT)
public record TubeOutput<E extends Throwable>(
    @NotNull BinaryResource<E> implementationModule,
    @NotNull BinaryResource<E> interfaceModule
) {}
